src/main.rs

```diff
@@ -132,33 +132,20 @@ fn setup(args: &[String]) -> Setup {
     let mixer_name = matches.opt_str("mixer");
     let mixer = mixer::find(mixer_name.as_ref())
         .expect("Invalid mixer");
-    let initial_volume;
-    // check if initial-volume argument is present
-    if matches.opt_present("initial-volume"){
-        // check if value is a number
-        if matches.opt_str("initial-volume").unwrap().parse::<i32>().is_ok(){
-            // check if value is in [0-100] range, otherwise put the bound values
-            if matches.opt_str("initial-volume").unwrap().parse::<i32>().unwrap() < 0 {
-                initial_volume = 0 as i32;
-            }
-            else if matches.opt_str("initial-volume").unwrap().parse::<i32>().unwrap() > 100{
-                initial_volume = 0xFFFF as i32;
-            }
-            // checks ok
-            else{
-                initial_volume = matches.opt_str("initial-volume").unwrap().parse::<i32>().unwrap()* 0xFFFF as i32 / 100 ;
+
+    let initial_volume: i32;
```

Comment:
```rust
let initial_volume = matches
    .opt_str("initial-volume")
    .map(|volume| {
        let volume = volume.parse::<i32>().unwrap();
        if volume < 0 || volume > 100 {
            panic!("Initial volume must be in the range 0-100");
        }
        volume * 0xFFFF / 100
    })
    .unwrap_or(0x8000);
````

Reply:
@plietar didn't see this before the merge. Map is neater than the if/else/matches statements, but I think panicking for a bad initial volume is a bit much? Perhaps print an info message and set to default volume?

Reply:
Similar question to what I had wrt. discovery mode: should we panic or fall back to something which might be surprising the user, but not totally kill the experience? In the discovery case you preferred the panic ;-).

Personally I'm all for reasonable defaults/fallbacks. But here as well as in the discovery case the surprise on the user's side could be bad: imagine he wanted to have a low volume, but for whatever reason used an invalid negative value. And then playback would start at 80%. Poor UX. It's debatable for sure.